napari/layers/_layer_actions.py

```diff
@@ -50,13 +50,16 @@ def _split_rgb(ll: LayerList):
 def _convert(ll: LayerList, type_: str):
     from napari.layers import Shapes
 
+    float_dtypes = [np.float64, np.float32, np.float16]
```

Comment:
This is potentially a future problem - what will happen if/when napari allows other array types besides numpy? (eg: cupy arrays, or some other type). I'm not entirely sure that this code will work in those future cases, or if there's a better way to write it here.

Reply:
NumPy defines `np.integer` and `np.floating` that could be used for checking with `issubclass` to check if data are in proper type.

Reply:
@GenevieveBuckley we already allow dask, zarr, and tensorstore arrays. I haven't looked at the code in this PR in detail but I am concerned... 😂  Since we have tensorstore and zarr (?) as test dependencies, I think we should add test parameters using those arrays where this code path is hit. Examples:

https://github.com/napari/napari/blob/0103bc4d360013efa64d389f0766954302160a07/napari/layers/labels/_tests/test_labels_key_bindings.py#L26

https://github.com/napari/napari/blob/0103bc4d360013efa64d389f0766954302160a07/napari/utils/_tests/test_dtype.py#L10

https://github.com/napari/napari/blob/0103bc4d360013efa64d389f0766954302160a07/napari/_qt/layer_controls/_tests/test_qt_image_base_layer_.py#L131